### PR TITLE
CC: Prepare secure execution image tarball for s390x

### PR DESCRIPTION
**Pull request by BbolroC**
This is to prepare a secure image tarball to run a confidential container for IBM Z SE(TEE).

Fixes: #6206

Signed-off-by: Hyounggyu Choi <Hyounggyu.Choi@ibm.com>

### CODE REVIEW

**Review by hbrueckner — APPROVED** [review]
Hi @BbolroC ! LGTM! Thanks a lot!

**Review by stevenhorsman — APPROVED** [review]
LGTM. Thanks @BbolroC!

File: tools/osbuilder/rootfs-builder/rootfs.sh [hunk 1]
```diff
@@ -657,9 +657,9 @@ EOF
 	touch "$dns_file"
 
     if [ -n "${AA_KBC}" ]; then
-		if [ "${AA_KBC}" == "offline_sev_kbc" ]; then
+		if [[ "${AA_KBC}" =~ ^offline_(sev|fs)_kbc$ ]]; then
 			info "Adding agent config for ${AA_KBC}"
-			AA_KBC_PARAMS="offline_sev_kbc::null" envsubst < "${script_dir}/agent-config.toml.in" | tee "${ROOTFS_DIR}/etc/agent-config.toml"
+			AA_KBC_PARAMS="${AA_KBC}::null" envsubst < "${script_dir}/agent-config.toml.in" | tee "${ROOTFS_DIR}/etc/agent-config.toml"
```
Comment (stevenhorsman) [inline tools/osbuilder/rootfs-builder/rootfs.sh:662]: Sorry to add this after previous approving, but I just wanted to check why this change was needed?
- If you want to set the `aa_kbc_params` we seem to have moved to using kernel_params as it's more flexible and updatable 
- If you want to restrict the endpoints I have some concerns about the brittleness of it and we've discussed this a little bit in https://github.com/kata-containers/kata-containers/issues/5590

Comment (BbolroC) [reply tools/osbuilder/rootfs-builder/rootfs.sh:662]: Yep, totally valid. I will resolve your concern. Thanks!

**Review by stevenhorsman — COMMENTED** [review]
@BbolroC - this update looks good removing the duplication. Once you are happy with it can I request you fixup the commits to clean the history please. Thanks!

**Review by fidencio — APPROVED** [review]
lgtm, thanks @BbolroC!

**Comment by BbolroC** [comment]
Why don't we merge this unless the test failures for tdx have something to do with this change?

**Comment by BbolroC** [comment]
@fidencio do you mind checking why the tests for tdx are failing? If you do not have a permission for it, could you point me out to a person who can check the log?

**Comment by stevenhorsman** [comment]
> @fidencio do you mind checking why the tests for tdx are failing? If you do not have a permission for it, could you point me out to a person who can check the log?

@arronwy - maybe you could help with this next week?

**Comment by fidencio** [comment]
The failure on the Cloud Hypervisor TDX test is related to network:
```
09:02:25 warning: spurious network error (1 tries remaining): [28] Timeout was reached (failed to download any data for `sha3 v0.10.6` within 30s)
09:02:55 error: failed to download from `[https://crates.io/api/v1/crates/sha3/0.10.6/download`](https://crates.io/api/v1/crates/sha3/0.10.6/download%60)
09:02:55 
09:02:55 Caused by:
09:02:55   [28] Timeout was reached (failed to download any data for `sha3 v0.10.6` within 30s)
```

**Comment by fidencio** [comment]
QEMU TDX:
```
22:00:31 # Mar 15 04:59:22 tdx-ci kata[1256856]: time="2023-03-15T04:59:22.680886829+08:00" level=error msg="agent pull image err. rpc error: code = Internal desc = failed to pull manifest Not authorized: url https://quay.io/v2/kata-containers/confidential-containers-auth/manifests/test" name=containerd-shim-v2 pid=1256856 sandbox=40a8fe4f2d607ea3488c62e6fed219a6692683e767374a2284bf122c005442a1 source=virtcontainers subsystem=kata_agent
```

cc @arronwy

**Comment by fidencio** [comment]
Cloud Hypervisor passed, but didn't repor back: http://10.112.240.208:8080/job/kata-containers-CCv0-cloud-hypervisor-TDX-CRI_CONTAINERD_K8S-PR/765/console

**Review by arronwy — APPROVED** [review]
LGTM, TDX Qemu CI error is not releated to this PR.

**Pull request merged**